src/pyramid_sanity/__init__.py

```diff
@@ -1,44 +1,19 @@
-"""Main integration into Pyramids automatic configuration hoovering."""
-
-from pyramid.config import PHASE3_CONFIG
-from pyramid.interfaces import ITweens
+from pyramid.tweens import MAIN
 
 from pyramid_sanity._egress import EgressTweenFactory
 from pyramid_sanity._ingress import IngressTweenFactory
 from pyramid_sanity._settings import SanitySettings
 
-__all__ = ["includeme"]
```

Comment:
Removed an unnecessary `__all__`

Reply:
All `__all__` statements are technically unnecessary, but I don't think we need to throw the baby out with the bath water here.

These were removed from other products because they were used to define and maintain _internal_ boundaries. It seems reasonable to define and maintain our _external_ boundary here as a library.

Reply:
Who's the `__all__` for / what purpose is it serving?

I don't think this package is actually a library, in that no one is going to write code to import and call this stuff. Instead, it's a Pyramid extension, and Pyramid just finds `includeme()`'s, doesn't need an `__all__` to do so. The plug point is already defined for us by the Pyramid framework.

I think this is similar to if our apps added all of their view functions to `__all__`'s: this is similarly an external boundary in that Pyramid imports and calls the view functions but the `__all__` would serve no purpose because Pyramid doesn't use it. (Or for that matter if our apps added all their `includeme()`'s to `__all__`'s.)

I could move this to a separate PR as there's some reorganising of the imports I wanted to do that might make this make more sense.

Off-topic discussion because I don't think `pyramid-sanity` is a library, but even in an actual library I think I'd just use the presence or lack of leading underscores and documentation to define the external boundary, not `__all__`. PEP 8 says that names without leading underscores are public, names with leading underscores are internal. Imported names (like `EgressTweenFactory` and `IngressTweenFactorry`) are internal by default according to PEP 8, but (again, says PEP 8) this can be overridden by documentation. So we have all the tools that we need for defining an external boundary, without `__all__`. There's nothing in PEP 8 that suggests that `__all__` is to be used to define external boundaries for human / documentation purposes. It just says that `__all__` is for modules that're designed to be used via `import *` and "to better support introspection", but I'm not aware of any tool that actually uses `__all__` for introspection (any IDE or autocomplete or documentation generator etc pretty much has to work without `__all__`'s I think, because most Python code doesn't keep up-to-date `__all__`'s everywhere). I just can't think of any solid reason to actually use `__all__`'s unless you're actually writing a module meant to be used with `import *`.

Reply:
I've put this `__all__` back for this PR, will send this and related organization changes in a separate PR so they can be discussed separately

Reply:
I think this might be too much discussion honestly. Pycharm uses it when doing auto imports, so it's handy to not import say "requests" or something from a random place, just because it got imported there. It's a very minor benefit. You seem to have a very strong opinion about it, and I don't, so lets go with not having it.

I think the library portion people are interested in is actually the exceptions, so it might make sense to do import them here for export, but we didn't.

Reply:
You wouldn't be writing code to import things from this Pyramid extension though, right? Anyway, sure let's go without it (but I  already moved the deletion to another PR)